query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapQuerySpec.groovy

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.ldap.discover
+
+import org.codice.ddf.admin.api.report.FunctionReport
+import org.codice.ddf.admin.common.fields.base.BaseFunctionField
+import org.codice.ddf.admin.common.fields.common.CredentialsField
+import org.codice.ddf.admin.common.fields.common.HostnameField
+import org.codice.ddf.admin.common.fields.common.PortField
+import org.codice.ddf.admin.common.report.message.DefaultMessages
+import org.codice.ddf.admin.ldap.LdapTestingCommons
+import org.codice.ddf.admin.ldap.TestLdapServer
+import org.codice.ddf.admin.ldap.commons.LdapMessages
+import org.codice.ddf.admin.ldap.fields.LdapDistinguishedName
+import org.codice.ddf.admin.ldap.fields.connection.LdapBindMethod
+import org.codice.ddf.admin.ldap.fields.connection.LdapBindUserInfo
+import org.codice.ddf.admin.ldap.fields.connection.LdapConnectionField
+import org.codice.ddf.admin.ldap.fields.connection.LdapEncryptionMethodField
+import org.codice.ddf.admin.ldap.fields.query.LdapQueryField
+import spock.lang.Specification
+
+import static org.codice.ddf.admin.ldap.LdapTestingCommons.noEncryptionLdapConnectionInfo
+import static org.codice.ddf.admin.ldap.LdapTestingCommons.simpleBindInfo
+
+class LdapQuerySpec extends Specification {
```

Comment:
I'd like to wait on this until merge, these constants are specific to the in memory LDAP that lives in DDF. We're temporarily copying it for now.